site/docs/v1/tech/lambdas/openid-connect-response-reconcile.adoc

```diff
@@ -45,12 +46,22 @@ If the JWT from the OIDC identity provider does not come back with an email clai
 
 If the Userinfo response available to you in the lambda has unique user information, you can build a fake email address from it.
 
-Here, the `sub` claim is the unique user id, and we're building an email address:
+Also, if the id_token from the token endpoint can be validated using the client secret then it will be available and may contain the email as well.
+
+Here, we check for an email in the idToken first, then fallback on using the `sub` claim as the unique user id, and build an email address:
 
 ```javascript
-function(user, registration, jwt) {
-  // The user's unique Id is the 'sub' claim.
-  user.email = jwt.sub + '@no-email-present.example.com';
+function(user, registration, jwt, idToken) {
+  // OR you can possibly get it from the idToken if available
+  if (idToken && idToken.email) {
+    console.info('email from id_token: ' + idToken.email);
+    user.email = idToken.email;
+  }
+  else {
+    // The user's unique Id is the 'sub' claim.
+    user.email = jwt.sub + '@no-email-present.example.com';
```

Comment:
We actually no longer allow you to modify the `email` in a lambda. We should double check that, but as far as I know this won't work. We should come up with a new "example" lambda that does not modify the `email`.